src/test/fixtures/mocks/window-history.js

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import { coerceMatchingShape } from '../../../utils/flow';
+
+/**
+ * jsdom leaves the history in place after every test, so the history will be dirty.
+ * Plus when we stop using a component it could still be subscribed to events, which
+ * does not really reflect real-world use cases. This mock creates a history API
+ * that can be thrown away after every use.
+ */
+export function mockWindowHistory() {
+  const originalHistory = Object.getOwnPropertyDescriptor(window, 'history');
+
+  let states = [null];
+  let index = 0;
+  const history = {
+    get length() {
+      return states.length;
+    },
+    scrollRestoration: 'auto',
+    state: null,
+    back() {
+      if (index <= 0) {
+        return;
+      }
+      index--;
+      history.state = states[index];
+      window.dispatchEvent(new Event('popstate'));
+    },
+    forward() {
+      if (index === states.length - 1) {
+        return;
+      }
+      index++;
+
+      history.state = states[index];
+      window.dispatchEvent(new Event('popstate'));
+    },
+    go() {
+      throw new Error('Not implemented.');
+    },
+    pushState(newState: any, _title: string, _url?: string) {
+      // The title and URL are ignored.
```

Comment:
maybe we should keep the current url in a local property? Or update window.location somehow? Not sure, we can keep that for when we'll need it too.